src/objectid.ts

```diff
@@ -37,8 +56,16 @@ export class ObjectId extends BSONValue {
 
   static cacheHexString: boolean;
 
-  /** ObjectId Bytes @internal */
-  private buffer!: Uint8Array;
+  /**
+   * The size of the buffer pool for ObjectId.
+   */
+  static poolSize: number = defaultPoolSize;
```

Comment:
> Would it be valuable to support setting poolSize = 0 which would essentially revert back to the original buffer-per-instance implementation?

Maybe I am missing something, would `ObjectId.poolSize = 1` do that, so could we sort of "clamp" to a minimum of 1?

IIUC, currently, 0 is going to create an empty pool, which will end up just ignoring input:

```js
let b = new Uint8Array(0)
b[0] = 1;
b[0] // undefined
```

Reply:
And by clamp, I think I would be fine with documenting the bounds, or we can do something fancy with a setter 🤔

Reply:
You're right, I was just overthinking it and if poolSize === 1 then we'll still perform some unnecessary `subArray(0,12)` in some cases. Which works fine.

What I ended up doing was adding that *fancy* getter/setter so poolSize must be > 0.